### PR TITLE
[RTC-345] Prefix env vars with JF

### PR DESCRIPTION
**Pull request by mickel8**
Let's follow good practices and perfix our env vars with `JF` not to overlap with other software. 

I also deleted `JELLYFISH_ADDRESS`. The idea is that `JF_HOST` is how we are seen by the others and it's used both for generating URLs and connecting to correct Jellyfish (i.e. address returned when we create a room).

### CODE REVIEW

**Review by sgfn — APPROVED** [review]
LGTM, don't forget to update the docs :pray:

File: config/runtime.exs [hunk 1]
```diff
@@ -11,62 +11,62 @@ alias Jellyfish.ConfigReader
...
-  integrated_turn_tcp_port: ConfigReader.read_port("INTEGRATED_TURN_TCP_PORT"),
+    ConfigReader.read_port_range("JF_INTEGRATED_TURN_PORT_RANGE") || {50_000, 59_999},
+  integrated_turn_tcp_port: ConfigReader.read_port("JF_INTEGRATED_TURN_TCP_PORT"),
   jwt_max_age: 24 * 3600,
-  output_base_path: System.get_env("OUTPUT_BASE_PATH", "jellyfish_output") |> Path.expand(),
-  address: System.get_env("JELLYFISH_ADDRESS") || "#{host}:#{port}",
-  metrics_ip: ConfigReader.read_ip("METRICS_IP") || {127, 0, 0, 1},
-  metrics_port: ConfigReader.read_port("METRICS_PORT") || 9568
+  output_base_path: System.get_env("JF_OUTPUT_BASE_PATH", "jellyfish_output") |> Path.expand(),
+  address: "#{host}:#{port}",
```
Comment (Rados13) [inline config/runtime.exs:52]: Why did you remove `JELLYFISH_ADDRESS` variable?

Comment (mickel8) [reply config/runtime.exs:52]: As in the PR description :stuck_out_tongue:

Comment (Rados13) [reply config/runtime.exs:52]: Are you sure that this won't break when you use for example proxy? Besides IMO it will look strange if somebody set `JF_HOST` to some domain and then jellyfish will send his clients to connect to `domain`:`PORT`.

Comment (mickel8) [reply config/runtime.exs:52]: Why? #{host}:#{port} was default in the previous implementation. That's also our HTTP server address and when we open WS connection we can use FQDN, we don't have to use ip address. We also don't use this for WebRTC stuff so I would try with this setup as it limits the number of env vars.

Comment (mickel8) [reply config/runtime.exs]: After discussing the problem offline, I added JF_HOST_PORT.

The problem was that without JF_HOST_PORT, we would return invalid port to Jellyfish Client .e.g 5002 instead of 443 (when running behind proxy) so we either have to allow to pass port in JF_HOST or add additional env var. I opt for additional env var as it is easier to understand and describe those 4 variables (JF_HOST, JF_HOST_PORT, JF_IP, JF_PORT). See .env.sample for more detailed explanation.

cc @sgfn @roznawsk

File: .env.sample [hunk 2]
```diff
@@ -1,19 +1,76 @@
...
+#
+#   -v $(pwd)/jellyfish_output:/app/jellyfish_output 
+#
+# JF_OUTPUT_BASE_PATH=/app/jellyfish_output
+
+
+# WEBRTC ENVS
+
+# true, if WebRTC peers are used
+JF_WEBRTC_USED=true
```
Comment (Rados13) [inline .env.sample:70]: Why do some variable defaults are commented out and some not? Shouldn't we simply provide default values for all variables where it is possible?

File: config/runtime.exs [hunk 3]
```diff
@@ -78,5 +85,5 @@ case System.get_env("SERVER_API_TOKEN") do
 end
 
 if prod? do
-  config :jellyfish, JellyfishWeb.Endpoint, url: [host: host, port: 443, scheme: "https"]
+  config :jellyfish, JellyfishWeb.Endpoint, url: [scheme: "https"]
```
Comment (Rados13) [inline config/runtime.exs:88]: Could we move all setting up configs about `JellyfishWeb.Endpoint` closes together and the second group would configs under key `:jellyfish`? By this I mean moving this if after `check_origin` if and moving `JF_SERVER_API_TOKEN` before `config :jellyfish, JellyfishWeb.Endpoint`.

File: .env.sample [hunk 4]
```diff
@@ -1,19 +1,76 @@
...
+# JF_IP and JF_PORT defines ip and port
+# our HTTP endpoint socket will listen to.
+# The default for Docker builds is 0.0.0.0 to allow 
+# access from the outside of container.
+# Note that this cannot be 127.0.0.1 as 
+# in Docker it only allows for traffic
+# from within the container.
+# For other builds, it is 127.0.0.1 not to 
+# accidentaly expose Jellyfish to the outside world
+# when it runs behind some proxy like nginx.
```
Comment (roznawsk) [inline .env.sample:10]:
```suggestion
# In Docker set it to 0.0.0.0 to allow 
# access from outside of the container.
# Note that this cannot be 127.0.0.1 as 
# in Docker it only allows for traffic
# from within the container.
# For other builds, set it to 127.0.0.1 to 
# avoid accidentally exposing Jellyfish to the outside world
# when it runs behind some proxy like nginx.
```

**Review by roznawsk — CHANGES_REQUESTED** [review]

**Review by sgfn — COMMENTED** [review]
Besides issues outlined by @roznawsk and @Rados13, LGTM

I agree that we should limit the comments in the `.env` files to an absolute minimum, 1-2 lines max per variable. Anything else we might need to tell the users about should be in the docs

File: .env.sample (hunk 2 above)
Comment (mickel8) [reply .env.sample:70]: Uncommented are the ones needed to be set to get JF running locally

File: docker-compose.yaml [hunk 5]
```diff
@@ -47,7 +47,8 @@ services:
       <<: *jellyfish-environment
       RELEASE_NODE: app@app2
       NODE_NAME: app@app2
-      PORT: 4002
+      JF_HOST: "localhost:4002"
```
Comment (mickel8) [inline docker-compose.yaml:50]: I am not sure if I am gonna introduce this change as IMO its misleading now :thinking:

File: .env.sample (hunk 2 above)
Comment (Rados13) [reply .env.sample]: `JF_WEBRTC_USED` isn't needed because you could use JF without WebRTC, so if `JF_WEBRTC_USED` is set `JF_OUTPUT_BASE_PATH` should be set also I think.

**Review by Rados13 — APPROVED** [review]
Besides your doubts, LGTM.

**Review by roznawsk — APPROVED** [review]
👍🏽 

I think it would be best if you introduced those changes to elixir and python SDK and then merge everything at the same time

File: docker-compose.yaml (hunk 5 above)
Comment (mickel8) [reply docker-compose.yaml:50]: If we had 4 vars we could get rid of `:4002` but let's try with 3 vars and see how it works

File: .env.sample (hunk 2 above)
Comment (mickel8) [reply .env.sample]: Yeah but I aimed at WebRTC usage as it's most popular
